### PR TITLE
[R-package] Updated lgb.Booster.R with keyword arguments

### PR DESCRIPTION
**Pull request by Pey-crypto**
-I am facing problems while generating the docs
Little help please
#3390

### CODE REVIEW

File: R-package/R/lgb.Booster.R [hunk 1]
```diff
@@ -335,7 +335,7 @@ Booster <- R6::R6Class(
 
       lower_bound <- 0.0
       lgb.call(
-        "LGBM_BoosterGetLowerBoundValue_R"
+        fun_name="LGBM_BoosterGetLowerBoundValue_R"
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:338]:
```suggestion
        fun_name = "LGBM_BoosterGetLowerBoundValue_R"
```

Can you please put a space between the argument name and value for all of these cases?

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks for taking the time to contribute to `{lightgbm}`!

There are other function calls in `lgb.Booster.R` that also should be using keyword arguments.

Can you please add keyword arguments to these calls in this file?

* `lgb.check.r6.class()`
* `lgb.params2str()`
* `private$inner_eval()`
* `lgb.call.return.str()`
* `predictor$predict()`
* `Predictor$new()`
* `self$save_model_to_string()`
* `object$predict()`
* `booster$save_model()`
* `booster$dump_model()`

Once you've made these changes, I can re-generate the documentation and push it to this pull request.

File: R-package/R/lgb.Booster.R (hunk 1 above)
Comment (Pey-crypto) [reply R-package/R/lgb.Booster.R:338]: Okay..will be done soon

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks! I left a few small suggestions

File: R-package/R/lgb.Booster.R [hunk 2]
```diff
@@ -496,7 +499,16 @@ Booster <- R6::R6Class(
...
+      predictor$predict(
+        data = data, 
+        start_iteration = start_iteration, 
+        num_iteration = num_iteration, 
+        rawscore = rawscore,
+        predleaf = predleaf, 
+        predcontrib = predcontrib, 
+        header = header, 
+        reshape = reshape
+        )
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:511]:
```suggestion
      predictor$predict(
          data = data
          , start_iteration = start_iteration
          , num_iteration = num_iteration
          , rawscore = rawscore
          , predleaf = predleaf
          , predcontrib = predcontrib
          , header = header
          , reshape = reshape
      )
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

File: R-package/R/lgb.Booster.R [hunk 3]
```diff
@@ -387,7 +387,10 @@ Booster <- R6::R6Class(
       }
 
       # Evaluate data
-      private$inner_eval(name, data_idx, feval)
+      private$inner_eval(
+        name = name,
+        data_idx = data_idx,
+        feval = feval)
```
[the remark this answers is not in the document]
Comment (jameslamb) [reply R-package/R/lgb.Booster.R]: I think accepting this and the other suggestions will fix the test failures on GitHub Actions. The right keyword argument here is `data_name`, not `name`

File: R-package/R/lgb.Booster.R [hunk 4]
```diff
@@ -873,7 +885,9 @@ lgb.save <- function(booster, filename, num_iteration = NULL) {
   }
 
   # Store booster
-  invisible(booster$save_model(filename, num_iteration))
+  invisible(booster$save_model(
+    filename = filename
+    , num_iterations = num_iteration))
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:890]:
```suggestion
    , num_iterations = num_iteration
  ))
```

File: R-package/R/lgb.Booster.R [hunk 5]
```diff
@@ -758,14 +770,14 @@ predict.lgb.Booster <- function(object,
...
   object$predict(
-    data
-    , start_iteration
-    , num_iteration
-    , rawscore
-    , predleaf
-    , predcontrib
-    , header
-    , reshape
+    data = data
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:773]:
```suggestion
      data = data
```

File: R-package/R/lgb.Booster.R [hunk 6]
```diff
@@ -387,7 +387,10 @@ Booster <- R6::R6Class(
       }
 
       # Evaluate data
-      private$inner_eval(name, data_idx, feval)
+      private$inner_eval(
+        data_name = name
+         , data_idx = data_idx
+         , feval = feval)
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:393]:
```suggestion
         , feval = feval
       )
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

File: R-package/R/lgb.Booster.R [hunk 7]
```diff
@@ -496,7 +500,16 @@ Booster <- R6::R6Class(
...
+      predictor$predict(
+          data = data
+          , start_iteration = start_iteration
+          , num_iteration = num_iteration
+          , rawscore = rawscore
+          , predleaf = predleaf
+          , predcontrib = predcontrib
+          , header = header
+          , reshape = reshape
+        )
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:512]:
```suggestion
      )
```

File: R-package/R/lgb.Booster.R [hunk 8]
```diff
@@ -873,7 +886,10 @@ lgb.save <- function(booster, filename, num_iteration = NULL) {
   }
 
   # Store booster
-  invisible(booster$save_model(filename, num_iteration))
+  invisible(booster$save_model(
+    filename = filename
+    , num_iterations = num_iteration
+    ))
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:892]:
```suggestion
  ))
```

File: R-package/R/lgb.Booster.R [hunk 9]
```diff
@@ -387,7 +387,11 @@ Booster <- R6::R6Class(
       }
 
       # Evaluate data
-      private$inner_eval(name, data_idx, feval)
+      private$inner_eval(
+        data_name = name
+         , data_idx = data_idx
+         , feval = feval
+         )
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:394]:
```suggestion
      )
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
sorry for all of the requested style changes. If you just click on the suggestions in the browser instead of committing them yourself, it might be easier

File: R-package/R/lgb.Booster.R [hunk 10]
```diff
@@ -873,7 +886,10 @@ lgb.save <- function(booster, filename, num_iteration = NULL) {
   }
 
   # Store booster
-  invisible(booster$save_model(filename, num_iteration))
+  invisible(booster$save_model(
+    filename = filename
+    , num_iterations = num_iteration
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:891]:
```suggestion
    , num_iteration = num_iteration
```

this parameter is called `num_iteration`

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for the efforts so far! Please see my suggested changes.

I'll review again once this PR also adds the keywords args for these:

* self$save_model_to_string()
* lgb.check.r6.class()
* Predictor$new()

File: R-package/R/lgb.Booster.R [hunk 11]
```diff
@@ -387,7 +387,11 @@ Booster <- R6::R6Class(
       }
 
       # Evaluate data
-      private$inner_eval(name, data_idx, feval)
+      private$inner_eval(
+        data_name = name
+         , data_idx = data_idx
+         , feval = feval
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:393]:
```suggestion
        , data_idx = data_idx
        , feval = feval
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

**Review by jameslamb — CHANGES_REQUESTED** [review]
looks good so far! But you are still missing these:

* `self$save_model_to_string()`
* `lgb.check.r6.class()`
* `Predictor$new()`

https://github.com/microsoft/LightGBM/pull/3496#pullrequestreview-519339738

If you're struggling with them, please let me know and I'd be happy to help!

**Comment by Pey-crypto** [comment]
Not sure about these 
-Predictor$new
-self$save_model_to_string

File: R-package/R/lgb.Booster.R [hunk 12]
```diff
@@ -39,12 +39,12 @@ Booster <- R6::R6Class(
         # Check if training dataset is not null
         if (!is.null(train_set)) {
           # Check if training dataset is lgb.Dataset or not
-          if (!lgb.check.r6.class(train_set, "lgb.Dataset")) {
+          if (!(lgb.Dataset = lgb.check.r6.class(train_set, "lgb.Dataset"))) {
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:42]: this code will result in overwriting the function `lgb.Dataset` with a boolean value. Please revert it by accepting this suggestion in the browser.
```suggestion
          if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset")) {
```

File: R-package/R/lgb.Booster.R [hunk 13]
```diff
@@ -216,7 +216,7 @@ Booster <- R6::R6Class(
       if (!is.null(train_set)) {
 
         # Check if training set is lgb.Dataset
-        if (!lgb.check.r6.class(train_set, "lgb.Dataset")) {
+        if (!(lgb.Dataset = lgb.check.r6.class(train_set, "lgb.Dataset"))) {
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:219]: Same suggestion here. This is not correct. Please accept this suggestion in the browser.

```suggestion
       if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset"))
```

File: R-package/R/lgb.Booster.R [hunk 14]
```diff
@@ -346,7 +346,7 @@ Booster <- R6::R6Class(
     eval = function(data, name, feval = NULL) {
 
       # Check if dataset is lgb.Dataset
-      if (!lgb.check.r6.class(data, "lgb.Dataset")) {
+      if (!(lgb.Dataset = lgb.check.r6.class(data, "lgb.Dataset"))) {
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:349]:
```suggestion
      if (!lgb.check.r6.class(object = data, name = "lgb.Dataset"))
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

**Comment by jameslamb** [comment]
> Not sure about these
> -Predictor$new
> -self$save_model_to_string

no problem! You can skip these, I will do them in a separate PR.

File: R-package/R/lgb.Booster.R [hunk 15]
```diff
@@ -216,7 +216,7 @@ Booster <- R6::R6Class(
       if (!is.null(train_set)) {
 
         # Check if training set is lgb.Dataset
-        if (!lgb.check.r6.class(train_set, "lgb.Dataset")) {
+       if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset"))
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:219]:
```suggestion
        if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset")) {
```

It looks like this `{` was lost, and the lines are no longer aligned.

File: R-package/R/lgb.Booster.R [hunk 16]
```diff
@@ -346,7 +346,7 @@ Booster <- R6::R6Class(
     eval = function(data, name, feval = NULL) {
 
       # Check if dataset is lgb.Dataset
-      if (!lgb.check.r6.class(data, "lgb.Dataset")) {
+      if (!lgb.check.r6.class(object = data, name = "lgb.Dataset"))
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:349]:
```suggestion
      if (!lgb.check.r6.class(object = data, name = "lgb.Dataset")) {
```

It looks like the `{` was lost

File: R-package/R/lgb.Booster.R [hunk 17]
```diff
@@ -149,7 +149,7 @@ Booster <- R6::R6Class(
     add_valid = function(data, name) {
 
       # Check if data is lgb.Dataset
-      if (!lgb.check.r6.class(data, "lgb.Dataset")) {
+      if (!(lgb.Dataset = lgb.check.r6.class(data, "lgb.Dataset"))) {
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:152]: this check is still incorrect. Please accept this suggestion in your browser.

```suggestion
      if (!lgb.check.r6.class(object = data, name = "lgb.Dataset")) {
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

File: R-package/R/lgb.Booster.R [hunk 18]
```diff
@@ -39,12 +39,12 @@ Booster <- R6::R6Class(
         # Check if training dataset is not null
         if (!is.null(train_set)) {
           # Check if training dataset is lgb.Dataset or not
-          if (!lgb.check.r6.class(train_set, "lgb.Dataset")) {
+          if (!lgb.check.r6.class(object = train_set,name = "lgb.Dataset")) {
```
Comment (jameslamb) [inline R-package/R/lgb.Booster.R:42]:
```suggestion
          if (!lgb.check.r6.class(object = train_set, name = "lgb.Dataset")) {
```

The linting jobs will complain about this missing space, sorry.

**Review by jameslamb — CHANGES_REQUESTED** [review]

File: R-package/R/lgb.Booster.R (hunk 18 above)
Comment (Pey-crypto) [reply R-package/R/lgb.Booster.R]: Thank you for constantly checking the commits, and also improving my code style.

**Pull request merged**